app/controllers/scim_rails/application_controller.rb

```diff
@@ -9,14 +9,30 @@ class ApplicationController < ActionController::API
     private
 
     def authorize_request
-      authenticate_with_http_basic do |username, password|
+      send(authentication_strategy) do |username, password|
```

Comment:
NITPICK: we should probably rename `username` to `searchable_attribute` to better express what the attribute is. `username` appears  to be a bit misleading.